### PR TITLE
#174797464 Reset Password

### PR DESCRIPTION
**Pull request by Adadie**
#### What does this PR do?

This PR implements the whole process for resetting the password. 

#### Description of Task to be completed?

- create forgot password UI.
- Create reset password UI.
- Validate both UI
- Create an action for forgot password.
- Create a reducer for forgot password.
- Create a reducer for reset password.

#### How should this be manually tested?

- Clone this repo `https://github.com/atlp-rwanda/script-heroes-frontend.git`.
- Check out this branch `ft-reset-password-#174797464`.
- Run `yarn`.
- Run `yarn start`.
- Register your email.
- Open this endpoint `/forgot`.
- Enter your email in the input box.
- Submit form
- Check if there is an email with reset password sent to you
- Click the reset button, it should redirect you to the reset password form.
- Fill in the form and submit.
- Try to login with the new password.
#### What are the relevant pivotal tracker stories?

[#174797464](https://www.pivotaltracker.com/story/show/174797464)
#### Screenshots (if appropriate)
Screen Shot 2020-12-14 at 16 57 07
Screen Shot 2020-12-14 at 17 07 30
Screen Shot 2020-12-14 at 17 07 42

### CODE REVIEW

**Review by elvisiraguha — COMMENTED** [review]
_.env_ file should not be pushed on the remote repo.

**Review by dushimeemma — APPROVED** [review]
Good job bro, but you have to work on css so that you can leave some space between main area and the footer and also on the form to enter new password you can work on it so that when the eye to view password is clicked can be checked else display as it is. also remove .env from remote.

**Comment by Adadie** [comment]
> Good job bro, but you have to work on css so that you can leave some space between main area and the footer and also on the form to enter new password you can work on it so that when the eye to view password is clicked can be checked else display as it is. also remove .env from remote.

Thank you @dushimeemma , though what do you mean by leaving space between main area!!?

File: redux/actions/forgotPassword.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import axios from "axios";
+import * as actions from "./types";
+
+const posting = axios.create({baseURL: `${process.env.REACT_APP_BACKEND_URL}`});
```
Comment (Habinezajanvier) [inline redux/actions/forgotPassword.js]: As all need this functionality, can't we make it global to prevent repeating ourselves

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by elvisiraguha — APPROVED** [review]

**Review by Habinezajanvier — APPROVED** [review]

**Review by valens10 — APPROVED** [review]

**Pull request merged**